### ISSUE TITLE
Support for collection cloning and copy documents between collections

### ISSUE THREAD

**Issue by hojabri** [typesense/typesense#1954]
## Description

As a user I would like to be able to clone a collection with or without the documents.
Also I would like to copy documents from an existing collection to a newly created one.

**Comment by jasonbosco** [comment]
You can already clone a collection without the documents: https://typesense.org/docs/27.1/api/collections.html#cloning-a-collection-schema

But yeah, agreed that it would be useful to be able to clone documents internally between collections.

**Comment by ajgeiss0702** [comment]
> This would be really usefull for backup and staging purpose. :)

I agree. I went to try to clone a collection for testing, then realized there was no way to do such a thing :(

**Comment by MantraMedia** [comment]
Are there any plans to add a clone functionality? 

This would provide a great workflow when updating the schema

-> index-{current} is an alias of index-{old_date}
-> clone index to {name}-{current_date}
-> apply schema changes
-> update alias of index-{current} to index-{current_date}

right now the documents need to reindexed manually or via a job all over again

**Comment by Clasyc** [comment]
> -> index-{current} points to the alias of index-{old_date}
-> clone index to {name}-{current_date}
-> apply schema changes
-> update alias of index-{current} to index-{current_date}

Yeah, I can imagine this is a really common procedure when using aliases. The fact that we have to manually insert documents into a new collection and reindex them again is inconvenient. I believe the Typesense team could make duplicating documents much easier by also copying existing indexes, avoiding the need for identical collection reindexing.

**Comment by Gabrimarin** [comment]
+1! Would be super useful for alias management

**Comment by tharropoulos** [comment]
On newer versions of Typesense (`v30.0.rca38+`), you can  copy documents when cloning collections by using `src_name` alongside `copy_documents` when creating a new collection.

[Here's](https://github.com/typesense/typesense/pull/2388) the PR with the demo.

And [here's](https://typesense.org/docs/30.0/api/collections.html#clone-collection-with-documents) the documentation.

**Comment by MantraMedia** [comment]
@tharropoulos  this does not support a workflow like above in which schema changes can be applied before the documents are reindexed , or ?

**Comment by tharropoulos** [comment]
No, this is only meant for copying the documents alongside the schema with a different name.
